### PR TITLE
[misc] move more workflows to self-hosted runners

### PR DESCRIPTION
**Pull request by bkmartinjr**
Move an additional 4 workflows to self-hosted runners

### CODE REVIEW

File: .github/workflows/py-dependency-check.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [single-cell-8c64g-runner, macos-latest]
```
Comment (beroy) [inline .github/workflows/py-dependency-check.yml:23]: Why does the runner appear in the `os` field?

File: .github/workflows/close-stale-prs.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
 
 jobs:
   close-issues:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, Linux, X64]
```
Comment (beroy) [inline .github/workflows/close-stale-prs.yml:8]: Just to be sure, in this PR you are not using large your intention is not to use `single-cell-8c64g-runner` or `single-cell-1tb-runner` runners.

**Review by beroy — APPROVED** [review]

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:23]: if you look a few lines below, these are the labels we provide to `runs-on`, which accepts a runner.  I suppose we could rename `os` to something better (e.g., `runner`?), but using `os` has been our convention for a while

File: .github/workflows/close-stale-prs.yml (hunk 2 above)
Comment (bkmartinjr) [reply .github/workflows/close-stale-prs.yml:8]: correct - this workflow can run on a much smaller runner. This set of labels is based on Jake's previous advice (see py-unittests.yml for an example)

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (beroy) [reply .github/workflows/py-dependency-check.yml:23]: I see it now

**Review by beroy — APPROVED** [review]

**Review by ebezzi — APPROVED** [review]

**Pull request merged**